src/lib/y2network/udev_rule.rb

```diff
@@ -0,0 +1,125 @@
+
+module Y2Network
+  # Simple udev rule class
+  #
+  # This class represents a network udev rule. The current implementation is quite simplistic,
+  # featuring a pretty simple API.
+  #
+  # @example Create a rule containing some key/value pairs (rule part)
+  #   rule = Y2Network::UdevRule.new(
+  #     Y2Network::UdevRulePart.new("ATTR{address}", "==", "?*31:78:f2"),
+  #     Y2Network::UdevRulePart.new("NAME", "=", "mlx4_ib3")
+  #   )
+  #   rule.to_s #=> "ACTION==\"add\", SUBSYSTEM==\"net\", ATTR{address}==\"?*31:78:f2\", NAME=\"eth0\""
+  #
+  # @example Create a rule from a string
+  #   rule = UdevRule.find_for("eth0")
+  #   rule.to_s #=> "ACTION==\"add\", SUBSYSTEM==\"net\", ATTR{address}==\"?*31:78:f2\", NAME=\"eth0\""
+  class UdevRule
+    class << self
+      # Returns the udev rule for a given device
+      #
+      # @param device [String] Network device name
+      # @return [UdevRule] udev rule
+      def find_for(device)
+        rules_map = Yast::SCR.Read(Yast::Path.new(".udev_persistent.net")) || {}
+        return nil unless rules_map.key?(device)
+        parts = rules_map[device].map { |p| UdevRulePart.from_string(p) }
+        new(parts)
+      end
+
+      # Helper method to create a rename rule based on a MAC address
+      #
+      # @param name [String] Interface's name
+      # @param mac  [String] MAC address
+      def new_mac_based_rename(name, mac)
+        new_network_rule([
+          UdevRulePart.new("ATTR{address}", "=", mac),
+          UdevRulePart.new("NAME", "=", name)
+        ])
+      end
+
+      # Helper method to create a rename rule based on the BUS ID
+      #
+      # @param name     [String] Interface's name
+      # @param mac      [String] BUS ID (e.g., "0000:08:00.0")
+      # @param dev_port [String] Device port
+      def new_bus_id_based_rename(name, bus_id, dev_port = nil)
+        parts = [UdevRulePart.new("KERNELS", "=", bus_id)]
+        parts.push(UdevRulePart.new("ATTR{dev_port}", "=", dev_port)) if dev_port
+        parts.push(UdevRulePart.new("NAME", "=", name))
+        new_network_rule(parts)
+      end
+
+      # Returns a network rule
+      #
+      # The network rule includes some parts by default.
+      #
+      # @param parts [Array<UdevRulePart] Additional rule parts
+      # @return [UdevRule] udev rule
+      def new_network_rule(parts = [])
+        base_parts = [
+          UdevRulePart.new("SUBSYSTEM", "==", "net"),
+          UdevRulePart.new("ACTION", "==", "add"),
+          UdevRulePart.new("DRIVERS", "==", "?*"),
+          UdevRulePart.new("ATTR{type}", "==", "1")
+        ]
+        new(base_parts.concat(parts))
+      end
+
+      # Writes udev rules to the filesystem
+      #
+      # @param udev_rules [Array<UdevRule>] List of udev rules
+      def write(udev_rules)
+        Yast::SCR.Write(Yast::Path.new(".udev_persistent.rules"), udev_rules.map(&:to_s))
+        Yast::SCR.Write(Yast::Path.new(".udev_persistent.nil"), [])
```

Comment:
explanation for this tricky call?

Reply:
Write changes to the file. We do something similar for sysconfig. I will add a note.

Reply:
I thought it looks differently with value nil and ".udev_persistent" as path, but I can be wrong.

Reply:
I expected that too, but I learn from the old code that, in this case, it is different. I tried it manually and the "expected" version does not work.